server/athenian/api/internal/settings.py

```diff
@@ -702,23 +713,48 @@ async def list_release_matches(
         """
         if repos is None:
             repos = await get_account_repositories(self._account, True, self._sdb)
-        repos = set(repos).union(coerce_prefixed_logical_repos(repos).keys())
+
+        # collect also physical repositories for requested logical repositories
+        repos_to_collect = set(repos).union(coerce_prefixed_logical_repos(repos).keys())
+
+        repo_names_to_collect: list[RepositoryName] = [
+            RepositoryName.from_prefixed(repo) for repo in repos_to_collect
+        ]
+        repo_names_to_collect_set = set(repo_names_to_collect)
```

Comment:
```suggestion
        repo_names_to_collect_set: set[RepositoryName] = {
            RepositoryName.from_prefixed(repo) for repo in repos_to_collect
        }
```
Does mypy require the hint?